plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/util/SymbolIcons.java

```diff
@@ -0,0 +1,108 @@
+                    put(SymbolKind.Field, JavaResources.INSTANCE.protectedField()),
+                    put(SymbolKind.Constructor, JavaResources.INSTANCE.protectedMethod()),
+                    put(SymbolKind.Enum, JavaResources.INSTANCE.enumItem()),
+                    put(SymbolKind.Interface, JavaResources.INSTANCE.interfaceItem()),
+                    put(SymbolKind.Field, JavaResources.INSTANCE.protectedField()),
+                    put(SymbolKind.Constant, JavaResources.INSTANCE.protectedField()),
+                    put(SymbolKind.EnumMember, JavaResources.INSTANCE.protectedField()))),
+            put(
+                Visibility.PACKAGE,
+                map(
+                    put(SymbolKind.Package, JavaResources.INSTANCE.packageItem()),
+                    put(SymbolKind.Class, JavaResources.INSTANCE.svgClassItem()),
+                    put(SymbolKind.Method, JavaResources.INSTANCE.defaultMethod()),
+                    put(SymbolKind.Field, JavaResources.INSTANCE.defaultField()),
+                    put(SymbolKind.Constructor, JavaResources.INSTANCE.defaultMethod()),
+                    put(SymbolKind.Enum, JavaResources.INSTANCE.enumItem()),
+                    put(SymbolKind.Interface, JavaResources.INSTANCE.interfaceItem()),
+                    put(SymbolKind.Field, JavaResources.INSTANCE.defaultField()),
+                    put(SymbolKind.Constant, JavaResources.INSTANCE.defaultField()),
+                    put(SymbolKind.EnumMember, JavaResources.INSTANCE.defaultField()))),
+            put(
+                Visibility.PRIVATE,
+                map(
+                    put(SymbolKind.Package, JavaResources.INSTANCE.packageItem()),
+                    put(SymbolKind.Class, JavaResources.INSTANCE.svgClassItem()),
+                    put(SymbolKind.Method, JavaResources.INSTANCE.privateMethod()),
+                    put(SymbolKind.Field, JavaResources.INSTANCE.privateField()),
+                    put(SymbolKind.Constructor, JavaResources.INSTANCE.privateMethod()),
+                    put(SymbolKind.Enum, JavaResources.INSTANCE.enumItem()),
+                    put(SymbolKind.Interface, JavaResources.INSTANCE.interfaceItem()),
+                    put(SymbolKind.Field, JavaResources.INSTANCE.privateField()),
+                    put(SymbolKind.Constant, JavaResources.INSTANCE.privateField()),
+                    put(SymbolKind.EnumMember, JavaResources.INSTANCE.privateField()))));
+  }
+
+  public SVGResource get(ExtendedSymbolInformation symbol) {
+    Visibility visiblity = symbol.getVisiblity();
+    return resources
+        .get(visiblity == null ? Visibility.PACKAGE : visiblity)
+        .get(symbol.getInfo().getKind());
+  }
+
+  private static <K, V> Consumer<Map<K, V>> put(K key, V value) {
+    return new Consumer<Map<K, V>>() {
+
+      @Override
+      public void accept(Map<K, V> map) {
+        map.put(key, value);
+      }
+    };
+  }
```

Comment:
If I'm not mistaken since java 8 we can simplify this 
 ```java
  private static <K, V> Consumer<Map<K, V>> put(K key, V value) {
    return new Consumer<Map<K, V>>() {
       @Override
      public void accept(Map<K, V> map) {
        map.put(key, value);
      }
    };
  }
 ```
 to this
 ```java  
  private static <K, V> Consumer<Map<K, V>> put(K key, V value) {
    return map -> map.put(key, value);
  }
 ``` 
or there is a requirement to create a new instance each time we call `put`?